src/main/java/com/minecolonies/coremod/commands/ActionArgumentType.java

```diff
@@ -0,0 +1,391 @@
+    }
+
+    public List<String> getTabCompletions(@NotNull final MinecraftServer server,
+            @Nullable final BlockPos pos,
+            final String potentialArgumentValue)
+    {
+        switch (this)
+        {
+            case INTEGER:
+                return Collections.emptyList();
+            case BOOLEAN:
+                return Arrays.asList(new String[] {"true", "false"});
+            case COORDINATE_X:
+            case COORDINATE_Y:
+            case COORDINATE_Z:
+                return getCoordinateTabCompletions(pos, potentialArgumentValue);
+            case PLAYER:
+                final List<String> playerNameStrings = getOnlinePlayerNames(server);
+                return playerNameStrings.stream().filter(k -> k.startsWith(potentialArgumentValue)).collect(Collectors.toList());
+            case COLONY:
+                return getColonyTabCompletions(potentialArgumentValue);
+            case CITIZEN:
+                return getCitizenTabCompletions(potentialArgumentValue);
+            default:
+                throw new IllegalStateException("Unimplemented ActionArgumentType tab completion");
+        }
+    }
+
+    private List<String> getCoordinateTabCompletions(@Nullable final BlockPos pos, final String potentialArgumentValue)
+    {
+        if (null == pos)
+        {
+            return Collections.emptyList();
+        }
+        if (potentialArgumentValue.isEmpty())
+        {
+            switch (this)
+            {
+                case COORDINATE_X:
+                    return Collections.singletonList(String.valueOf(pos.getX()));
+                case COORDINATE_Y:
+                    return Collections.singletonList(String.valueOf(pos.getY()));
+                case COORDINATE_Z:
+                    return Collections.singletonList(String.valueOf(pos.getZ()));
+                default:
+                    // We will never reach here.
+                    break;
+            }
+            return Collections.emptyList();
+        }
+        try
+        {
+            Integer.parseInt(potentialArgumentValue);
+            return Collections.emptyList();
+        }
+        catch (final NumberFormatException e)
+        {
+            return Collections.emptyList();
+        }
+    }
+
+    private List<String> getColonyTabCompletions(final String potentialArgumentValue)
+    {
+        final List<String> colonyNumberStrings = getColonyIdStrings();
+        if (potentialArgumentValue.isEmpty())
+        {
+            return colonyNumberStrings;
+        }
+        try
+        {
+            Integer.parseInt(potentialArgumentValue);
+            return colonyNumberStrings.stream().filter(k -> k.startsWith(potentialArgumentValue)).collect(Collectors.toList());
+        }
+        catch (final NumberFormatException e)
+        {
+            return Collections.emptyList();
+        }
+    }
+
+    private List<String> getCitizenTabCompletions(final String potentialArgumentValue)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this method to reduce its Cognitive Complexity from 33 to the 20 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)